src/main/scala/treadle/stage/phases/GetFirrtlAst.scala

```diff
@@ -0,0 +1,33 @@
+// See LICENSE for license details.
+
+package treadle.stage.phases
+
+import firrtl.annotations.Annotation
+import firrtl.{AnnotationSeq, Parser}
+import firrtl.options.Phase
+import firrtl.stage.FirrtlSourceAnnotation
+import treadle.TreadleCircuitAnnotation
+import treadle.executable.TreadleException
+
+import scala.collection.mutable
+
+object GetFirrtlAst extends Phase {
+  override def transform(annotationSeq: AnnotationSeq): AnnotationSeq = {
+    var sourcesFound = new mutable.ArrayBuffer[Annotation]()
+
+    val newAnnotationSeq = annotationSeq.flatMap {
+      case fs @ FirrtlSourceAnnotation(firrtlText) =>
+        sourcesFound += fs
+        val circuit = Parser.parse(firrtlText)
+        Some(TreadleCircuitAnnotation(circuit))
+
+      case other => Some(other)
+    }
+
+    if(sourcesFound.length > 1) {
+      throw TreadleException(s"Multiple sources of firrtl found for Treadle" + sourcesFound.mkString("\n", "\n", ""))
+    }
```

Comment:
Interesting... I generally put all the checking in one phase. However, this is a perfectly viable alternative and perhaps puts information closer to where it makes sense. (I say leave this as is.)

Reply:
For this check, should this be checking on multiple `TreadleCircuitAnnotation`s additionally or alternatively? What would happen if a user a passes in a `TreadleCircuitAnnotation` and a `FirrtlSourceAnnotation`?